pkg/ingester-rf1/ingester.go

```diff
@@ -539,10 +519,16 @@ func (i *Ingester) loop() {
 	flushTicker := util.NewTickerWithJitter(i.cfg.FlushCheckPeriod, j)
 	defer flushTicker.Stop()
 
+	jit := i.cfg.StreamRetainPeriod / 5
+	streamRetentionTicker := util.NewTickerWithJitter(i.cfg.StreamRetainPeriod, jit)
+	defer streamRetentionTicker.Stop()
+
 	for {
 		select {
 		case <-flushTicker.C:
 			i.doFlushTick()
+		case <-streamRetentionTicker.C:
```

Comment:
do you want to move this ticker check to a separate go routine loop so that `doFlushTick` would never be blocked by the stream cleanup? I imagine flushing data is more important than clearing memory at least in the short term

Reply:
Yeah, thats a good point. I'll create a separate goroutine

Reply:
+1 because I want to remove this select :)